### DISCUSSION TITLE
Wagmi package error when running test with Vitest

### DISCUSSION THREAD

**Discussion by Akaryatrh** [wevm/wagmi#2012] [Q&A] [answered]
Hi!

I'm trying to setup my config for unit testing. I'm running a react app built with Vite and trying to run tests with Vitest. Unfortunately, my test is failing with the following error:
```shell
Error: Missing "./dist/chains" specifier in "wagmi" package
 ❯ e node_modules/vite/dist/node/chunks/dep-ca21228b.js:14845:25
 ❯ n node_modules/vite/dist/node/chunks/dep-ca21228b.js:14845:627
 ❯ o node_modules/vite/dist/node/chunks/dep-ca21228b.js:14845:1276
 ❯ resolveExports node_modules/vite/dist/node/chunks/dep-ca21228b.js:22033:20
 ❯ resolveDeepImport node_modules/vite/dist/node/chunks/dep-ca21228b.js:22052:31
 ❯ tryNodeResolve node_modules/vite/dist/node/chunks/dep-ca21228b.js:21737:20
 ❯ Context.resolveId node_modules/vite/dist/node/chunks/dep-ca21228b.js:21488:28
 ❯ Object.resolveId node_modules/vite/dist/node/chunks/dep-ca21228b.js:41612:46
 ❯ TransformContext.resolve node_modules/vite/dist/node/chunks/dep-ca21228b.js:41363:23
```

My testing setup is highly inspired from wagmi testing setup here: https://github.com/wagmi-dev/wagmi/blob/main/packages/react/test/index.tsx

I succeeded to reproduce the issue on a stackblitz env: https://stackblitz.com/edit/vitest-dev-vitest-clnt15?file=src/main.tsx&view=editor (I tested with wagmi 0.10.0, 0.11.5 and 0.12.0 and ended up with the same error)

Also the `./dist/chains` is probably coming from the package exports defined here: https://github.com/wagmi-dev/wagmi/blob/main/packages/react/package.json#L41

Any help is greatly appreciated 🙂

**Comment by tmm** [comment, marked as answer]
You are using the wrong import entrypoint. Should be `'wagmi/chains'` not `'wagmi/dist/chains'`. Worked for me once I fixed that!

**Comment by Akaryatrh** [comment]
Oh my, this is so embarrassing 😅
Thx a lot for your help @tmm and at least the provided example will maybe help some people in the future!
